src/aioquic/quic/congestion/__init__.py

```diff
@@ -0,0 +1,145 @@
+                    self._filtered_min = self._sample_max
+
+                delta = self._sample_min - self._filtered_min
+                if delta * 4 >= self._filtered_min:
+                    self._increases += 1
+                    if self._increases >= self._size:
+                        return True
+                elif delta > 0:
+                    self._increases = 0
+        return False
+
+class QuicCongestionControl:
+
+    def __init__(self, *, max_datagram_size : int, callback=None, fixed_cwnd = 10*1024*1024) -> None:
+        self.callback = callback # a callback argument that is called when an event occurs
+        # 10 GB window or custom fixed size window (shouldn't be used in real network !, use a real CCA instead)
+        self._max_datagram_size = max_datagram_size
+        self.cwnd = fixed_cwnd
+        self.data_in_flight = 0
+
+    def set_recovery(self, recovery):
+        # recovery is a QuicPacketRecovery instance
+        self.recovery = recovery
+
+    def on_packet_acked(self, packet: QuicSentPacket, now : float):
+        if self.callback:
+            self.callback(CongestionEvent.ACK, self)
+        if type(self) == QuicCongestionControl:
+            # don't call this if it is a superclass that runs
+            self.data_in_flight -= packet.sent_bytes
+
+    def on_packet_sent(self, packet: QuicSentPacket, now : float) -> None:
+        if self.callback:
+            self.callback(CongestionEvent.PACKET_SENT, self)
+        if type(self) == QuicCongestionControl:
+            # don't call this if it is a superclass that runs
+            self.data_in_flight += packet.sent_bytes
+
+    def on_packets_expired(self, packets: Iterable[QuicSentPacket]) -> None:
+        if self.callback:
+            self.callback(CongestionEvent.PACKET_EXPIRED, self)
+        if type(self) == QuicCongestionControl:
+            for packet in packets:
+                # don't call this if it is a superclass that runs
+                self.data_in_flight -= packet.sent_bytes
+
+    def on_packets_lost(self, packets: Iterable[QuicSentPacket], now: float) -> None:
+        if self.callback:
+            self.callback(CongestionEvent.PACKET_LOST, self)
+        if type(self) == QuicCongestionControl:
+            for packet in packets:
+                # don't call this if it is a superclass that runs
+                self.data_in_flight -= packet.sent_bytes
+
+    def on_rtt_measurement(self, latest_rtt: float, now: float) -> None:
+        if self.callback:
+            self.callback(CongestionEvent.RTT_MEASURED, self)
+
+    def get_congestion_window(self) -> int:
+        # return the cwnd in number of bytes
+        return self.cwnd   
+    
+    def _set_congestion_window(self, value):
+        self.cwnd = value
+
+    def get_ssthresh(self) -> Optional[int]: 
+        pass
+
+    def get_bytes_in_flight(self) -> int:
+        return self.data_in_flight
+
+    def log_callback(self) -> Dict[str, Any]:
```

Comment:
This doesn't look like it's actually a "callback", it looks more like `get_log_data` or something?

Reply:
It is indeed used to get the logging data for the congestion controller, thus I renamed it